src/Tuttifrutti/Models/PaperCode.hs

```diff
@@ -3,20 +3,26 @@ module Tuttifrutti.Models.PaperCode where
 
 import           Tuttifrutti.Prelude
 
+import           Data.Aeson          (Value (String), withText)
 import qualified Data.Text           as Text
-import           Data.Unjson
+import           Data.Unjson         (Unjson (..), unjsonAeson)
 import           Database.Persist.TH (derivePersistField)
 
 data PaperCode
   = HBL
   | ON
   | VN
   | HT
+  | UnknownPaperCode Text
   deriving (Show, Eq, Generic, Read, Data, Ord)
 derivePersistField "PaperCode"
 
-instance FromJSON PaperCode
-instance ToJSON PaperCode
+instance FromJSON PaperCode where
+  parseJSON = withText "PaperCode" (pure . toPaperCode)
+instance ToJSON PaperCode where
+  toJSON c = case c of
```

Comment:
I am not sure about this implementation:  if we let a delivery reclamation be created with an unknown paper code, then I don't think we can fail when we encode it back to JSON. This is what this code reflects.

But do we always want this behaviour?